### PR TITLE
Add Scala-specific dev environment

### PR DESCRIPTION
**Pull request by kubukoz**

### CODE REVIEW

**Review by cole-h — DISMISSED** [review]

File: src/pages/start/3.nix-build-nixpkgs.mdx [hunk 1]
```diff
@@ -102,6 +102,16 @@ Let's build and run [ripgrep]:
 nix build "nixpkgs#ripgrep"
 ./result/bin/rg --help
 ```
+
+</SpecificLanguage>
+<SpecificLanguage client:load lang="Scala">
+Let's build and run [scalafmt]:
```
[the remark this answers is not in the document]
Comment (kubukoz) [reply src/pages/start/3.nix-build-nixpkgs.mdx:108]: yeah, this was a copy-paste miss :)

**Comment by lucperkins** [comment]
@kubukoz Thanks a ton for this! The only thing "wrong" with it is that it needs a package build example, as the language selector is also present in docs like [this](https://zero-to-nix.com/start/nix-build-nixpkgs). I'm going to PR a package build example against this PR and we can go from there!

**Comment by kubukoz** [comment]
isn't that what the scalafmt in [this doc](src/pages/start/3.nix-build-nixpkgs.mdx) is for? Unless I'm missing something

**Comment by lucperkins** [comment]
@kubukoz Basically, in addition to development environment templates, we also have package build templates for each language, which you can see [here](https://github.com/DeterminateSystems/zero-to-nix/tree/main/nix/templates/pkg). So for each language we provide both. But don't worry, I'm going to provide a package build example to complement this 😄

**Comment by kubukoz** [comment]
ah, I see. Might be a problem with Scala, given there are no builders for it in nixpkgs.

**Comment by lucperkins** [comment]
@kubukoz There aren't, but I don't see any reason not to use a third-party builder here (the approach I'm currently working on). The beauty of flakes is that it's easy to use third-party stuff and this seems like a good way to emphasize that.

**Comment by kubukoz** [comment]
That's absolutely fine by me. Let me know when you have something to review, packaging in Scala has been on my radar for a while.

**Comment by lucperkins** [comment]
@kubukoz Good news! I got a Scala package figured out. Once this passes CI I'm going to merge it and then merge my PR. Thanks a ton for your help 🥳 Really cool to have Scala support in Zero to Nix.

**Pull request merged**